htlcswitch/switch.go

```diff
@@ -0,0 +1,599 @@
+package htlcswitch
+
+import (
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"crypto/sha256"
+
+	"github.com/go-errors/errors"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/lightningnetwork/lnd/routing"
+	"github.com/roasbeef/btcd/wire"
+	"github.com/roasbeef/btcutil"
+)
+
+var (
+	// ErrChannelLinkNotFound is used when channel link hasn't been found.
+	ErrChannelLinkNotFound = errors.New("channel link not found")
+)
+
+// ChannelCloseType is a enum which signals the type of channel closure the
+// peer should execute.
+type ChannelCloseType uint8
+
+const (
+	// CloseRegular indicates a regular cooperative channel closure
+	// should be attempted.
+	CloseRegular ChannelCloseType = iota
+
+	// CloseForce indicates that the channel should be forcefully closed.
+	// This entails the broadcast of the commitment transaction directly on
+	// chain unilaterally.
+	CloseForce
+
+	// CloseBreach indicates that a channel breach has been dtected, and
+	// the link should immediately be marked as unavailable.
+	CloseBreach
+)
+
+// ChanClose represents a request wto close a particular channel specified by
+// its outpoint.
+type ChanClose struct {
+	CloseType ChannelCloseType
```

Comment:
This struct is missing `godoc` comments on all its attributes. 

Still working my way through this file, but I suspect this might not need to be exposed.